index.bs

```diff
@@ -94,10 +109,60 @@ Note: For an [=endpoint node=] the hostname in the above steps will be
 typically be "<code>localhost</code>".
 
 <div algorithm>
-To <dfn>handle an incoming message</dfn> given a
-[=WebSocket connection=] <var ignore>connection</var>:
+To <dfn>handle an incoming message</dfn> given a [=WebSocket connection=]
+|connection|, type |type| and data |data|:
+
+ 1. If |type| is not [=%x1 denotes a text frame|text=], return.
+
+    Issue: Should we instead close |connection| with [=status
+    codes|status code=] 1003, or [=respond with an error=]?
+
+ 2. [=Assert=]: |data| is a [=scalar value string=], because the
+     WebSocket [=handling errors in UTF-8-encoded data=] would already
+     have [=fail the WebSocket connection|failed the WebSocket
+     connection=] otherwise.
+
+    Issue: Nothing seems to define what [=status codes|status code=]
+    is used for UTF-8 errors.
+
+ 3. Let |parsed| be the result of [=parse JSON into Infra
+    values|parsing JSON into Infra values=] given |data|. If this
+    throws an exception, then [=respond with an error=] given
+    |connection| and [=error code=] [=invalid argument=], and finally
+    return.
+    <!-- corresponds to Parse error (-32700) in JSON-RPC -->
+
+ 4. If any of the following conditions are false:
+
+     1. |parsed| is a [=map=]
+
+     2. |parsed|["<code>id</code>"] <a for=map>exists</a> and is an
+        integer in the range [−2147483648, 2147483647].
```

Comment:
I think we need *some* range, so I probed CDP to see when it would fail due to the ID and not, and found this range, which is the same as https://heycam.github.io/webidl/#idl-long, or int32.

Note that "any non-negative integer that can be represented in JS" is not possible to fit even in an uint64. Experimentally, 18446744073709556000 is the smallest integer that can be represented as a double but not uint64.

I'll add an inline issue for this, also explaining where this came from.

Reply:
I was thinking of `2**53 - 1` i.e. the largest number such that `N` and `N + 1` both have exact representations in js. That obviously would fit into a [u]int64.

I definitely feel like negative ids are confusing. I wonder if the connection should have a concept of last seen id and be able to assert that the command has a higher id than the previous maximum one, or whether there are situations in which it's hard for the client to maintain that invariant.

Reply:
I've disallowed negative IDs and made the issue more elaborate.